### PR TITLE
Add spoilers to escapeMarkdown and update strikeout matching

### LINKED ISSUE #3048: escapeMarkdown does not escape spoiler syntax in messages
**Please describe the problem you are having in as much detail as possible:**

The newly introduced spoiler tag syntax is not yet handled in Util.escapeMarkdown. For example, `||bla||` would show up exactly like that before and now appears as a spoiler around `bla`, even after running it through escapeMarkdown().

**Include a reproducible code sample here, if possible:**

```js
console.log(Discord.escapeMarkdown('||test||'))

// expected: \|\|test\|\| or \||test\|| (both work)
// actually printed: ||test||
```

**Further details:**
- discord.js version: 11.4.2

Not tested on latest master, but the code looks like it doesn't handle pipes at all: https://github.com/discordjs/discord.js/blob/42505b78c15306491ac3a605f86a3e010bc80958/src/util/Util.js#L84

### PR DESCRIPTION
**Pull request by Darqam**
**Please describe the changes this PR makes and why it should be merged:**
Since spoilers have just been added with the `|| test||` syntax, add that to the escapeMarkdown function.

The same changes could be added to 11.4-dev as well I imagine.

edit: This also forces the regex to match two ~ in order to capture strikeouts instead of just a single one
This produces then
`\~~test\~~` and `\||test\||` instead of the previous `\~\~test\~\~` and `\|\|test\|\|`

Resolves #3048

**Status**
- [x] Code changes have been tested against the Discord API, or there are no code changes
- [ ] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**  
- [x] This PR changes the library's interface (methods or parameters added)
  - [x] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/util/Util.js [hunk 1]
```diff
@@ -79,7 +79,7 @@ class Util {
   static escapeMarkdown(text, onlyCodeBlock = false, onlyInlineCode = false) {
```
Comment (izexi) [inline src/util/Util.js]: Perhaps add a parameter `onlySpoiler` for this so it's consistent with the other two?

Comment (Darqam) [reply src/util/Util.js]: I imagined there was a reason not to, due to there not being `onlyStrikeout` or `onlyUnderline`, etc

Comment (DUFC-Liam) [reply src/util/Util.js]: In all honesty `onlySpoiler` would be useful.

**Comment by sauerbraten** [comment]
bump

**Comment by sauerbraten** [comment]
By the way, how does this have a "semver: patch" label when there are breaking changes?

**Comment by SpaceEEC** [comment]
Implemented with #3241.

**Pull request closed without merging**